### PR TITLE
Cindy start

### PR DESCRIPTION
**Pull request by cindyciclta**

### CODE REVIEW

File: src/chirp_client.cc [hunk 1]
```diff
@@ -105,10 +107,10 @@ int ChirpClient::follow(const std::string& user,
   }
 }
 
-int ChirpClient::read(const std::string& chirp_id) {
+int ChirpClient::read(const std::string &chirp_id) {
```
Comment (barath) [inline src/chirp_client.cc:110]: Minor: I'd recommend having reformatting be a separate pull request.  But it's fine.

File: src/chirp_client.cc [hunk 2]
```diff
@@ -8,10 +9,11 @@ DEFINE_string(read, "", "Reads the chirp thread starting at the given id");
 DEFINE_string(follow, "", "Starts following the given username");
 DEFINE_bool(monitor, false, "Streams new tweets from those currently followed");
 
-int ChirpClient::registeruser(const std::string& user) {
+int ChirpClient::registeruser(const std::string &user) {
+  std::cout << "here1" << std::endl;
```
Comment (barath) [inline src/chirp_client.cc]: Maybe use log statements instead?

File: src/chirp_client.cc [hunk 3]
```diff
@@ -140,18 +142,20 @@ int ChirpClient::read(const std::string& chirp_id) {
...
   grpc::ClientContext context;
   chirp::MonitorRequest request;
-  chirp::MonitorReply reply;  // Data from service layer will be updated here
+  chirp::MonitorReply reply; // Data from service layer will be updated here
 
   request.set_username(user);
   // error check and give info to user about followers
   if (!CheckMonitorInfo(user)) {
-    return 1;  // return if monitoring check failed and monitoring will fail
+    return 1; // return if monitoring check failed and monitoring will fail
```
Comment (barath) [inline src/chirp_client.cc:154]: What does 1 mean?

File: src/chirp_client.cc [hunk 4]
```diff
@@ -266,11 +270,23 @@ void ChirpClient::PrintChirpThread(
   std::cout << thread_string << std::endl;
   LOG(INFO) << thread_string << std::endl;
 }
+// Cindy's Implementation
+bool ChirpClient::stream(const std::string &hashtagword) {
+  grpc::ClientContext context;
+  chirp::StreamRequest request;
+  request.set_hashtag(hashtagword);
+  std::cout << "comes streams" << std::endl;
```
Comment (barath) [inline src/chirp_client.cc]: Logging?

**Comment by barath** [comment]
You should also remove all the non-code files from your pull request.

**Comment by barath** [comment]
You should also remove all the non-code files from your pull request.

File: src/service.cc [hunk 5]
```diff
@@ -6,6 +6,7 @@ ServiceLayer::ServiceLayer(KeyValueStoreInterface *kvs_connection) {
 // Constructs a RegisterRequest and sends to service layer thru grpc and
 // receives a RegisterReply back
 bool ServiceLayer::registeruser(const std::string &username) {
+  std::cout << "here3" << std::endl;
```
Comment (apetranik) [inline src/service.cc]: are these just to test or are you going to keep these?

File: src/service_controller.cc [hunk 6]
```diff
@@ -1,9 +1,11 @@
...
 
-grpc::Status ServiceController::registeruser(
-    grpc::ServerContext *context, const chirp::RegisterRequest *request,
-    chirp::RegisterReply *reply) {
+grpc::Status
+ServiceController::registeruser(grpc::ServerContext *context,
+                                const chirp::RegisterRequest *request,
+                                chirp::RegisterReply *reply) {
   // check if user is already registered
+  std::cout << "here2" << std::endl;
```
Comment (apetranik) [inline src/service_controller.cc]: same here

File: src/service_controller.cc [hunk 7]
```diff
@@ -103,9 +105,11 @@ grpc::Status ServiceController::validate_monitor_request(
...
 }
 
-grpc::Status ServiceController::monitor(
-    grpc::ServerContext *context, const chirp::MonitorRequest *request,
-    grpc::ServerWriter<chirp::MonitorReply> *stream) {
+grpc::Status
+ServiceController::monitor(grpc::ServerContext *context,
+                           const chirp::MonitorRequest *request,
+                           grpc::ServerWriter<chirp::MonitorReply> *stream) {
+  std::cout << "check monitor 2" << std::endl;
```
Comment (apetranik) [inline src/service_controller.cc]: same here

**Review by apetranik — APPROVED** [review]
Looks good! Next commit I would remove the binaries

**Comment by cindyciclta** [comment]
I accidentally kept my debugging statements. I reminded myself to remove them before I pushed but I forgot about it. I pushed the updated ones now~

**Comment by cindyciclta** [comment]
I realized I wasn't supposed to merge until I got both approvals. I'm sorry.

**Pull request merged**